### PR TITLE
chore: sync the latest code from main repo

### PR DESCRIPTION
**Pull request by LucasXu0**

### CODE REVIEW

File: lib/src/service/selection_service.dart [hunk 1]
```diff
@@ -337,7 +349,8 @@ class _AppFlowySelectionState extends State<AppFlowySelection>
     // try to select the word.
     final selection = currentSelection.value;
     if (selection == null ||
-        (selection.isCollapsed && currentSelectedNodes.first is TextNode)) {
+        (selection.isCollapsed == true &&
```
Comment (Xazin) [inline lib/src/service/selection_service.dart:352]: boolean conditionals don't need to be compared to another boolean `boolean == true`

File: lib/src/service/selection_service.dart [hunk 2]
```diff
@@ -700,4 +713,15 @@ class _AppFlowySelectionState extends State<AppFlowySelection>
...
   }
+
+  final List<SelectionInterceptor> _interceptors = [];
+  @override
+  void register(SelectionInterceptor interceptor) {
+    _interceptors.add(interceptor);
+  }
+
+  @override
+  void unRegister(SelectionInterceptor interceptor) {
```
Comment (Xazin) [inline lib/src/service/selection_service.dart:724]: unregister is one word

File: pubspec.yaml [hunk 3]
```diff
@@ -24,10 +24,10 @@ dependencies:
   url_launcher: ^6.1.5
   logging: ^1.0.2
   intl_utils: ^2.7.0
-  intl: ^0.17.0
+  intl: any
```
Comment (Xazin) [inline pubspec.yaml:27]: I'm curious why `any` would be preferred over a version?

**Comment by LucasXu0** [comment]
Oh Sorry. This is a test PR. I will close it.

**Pull request closed without merging**